src/amo/components/Search/index.js

```diff
@@ -60,49 +77,66 @@ export class SearchBase extends React.Component {
       LinkComponent,
       count,
       enableSearchSort,
+      errorHandler,
       filters,
-      hasSearchParams,
       loading,
-      page,
+      paginationQueryParams,
       pathname,
       results,
     } = this.props;
-    const queryParams = this.props.queryParams ||
+
+    const page = parsePage(filters.page);
+
+    // If paginator params aren't specified, we convert filters instead.
+    const queryParams = paginationQueryParams ||
       convertFiltersToQueryParams(filters);
-    const paginator = count > 0 && hasSearchParams ? (
+
+    // TODO: Make the count need to be higher than one page's worth of
+    // add-ons.
```

Comment:
It should be a separate bug, I'll file. It would be nice not to show the paginator if there are 23 add-ons (eg both prev/next pages will be disabled) but the `page_size` is 25.